src/api/route-services/work-response.js

```diff
@@ -15,6 +21,28 @@ class WorkResponseService {
     })();
   }
 
+  async notifyDataUpdate(responseForClient) {
+    const { experimentId } = responseForClient.request;
+
+    // this should send the request so the UI can make it again if needed
+    // instead of sending the data to everyone
+    const response = {
+      response: responseForClient,
+      status: NEW,
+      type: DATA_UPDATE,
+    };
+
+    // for now we only want to notify about cell sets updates
+    if (responseForClient.request.body.name === 'ClusterCells') {
+      const cellSets = JSON.parse(responseForClient.results[0].body);
+      await (new ExperimentService()).updateLouvainCellSets(experimentId, cellSets);
```

Comment:
Isn't it better to initialize `new ExperimentService` outside of the function? That way we can only have 1 instance of the object for the whole lifecycle of the process.